analyses/focal-cn-file-preparation/05-define-most-focal-cn-units.Rmd

```diff
@@ -0,0 +1,306 @@
+    by = c("biospecimen_id" = "Kids_First_Biospecimen_ID",
+           "chromosome_arm")
+  ) %>%
+  left_join(
+    consensus_seg_cytoband_status,
+    by = c("biospecimen_id" = "Kids_First_Biospecimen_ID",
+           "cytoband")
+  ) %>%
+  # Filter the annotated CN data to include only neutral arms and cytobands,
+  # and disagreements
+  filter(
+    # Case 1) Gene call disagrees with both arm and cytoband (This captures most
+    # we want to keep, including all of when arm and cytoband are neutral, uncallable
+    # or unstable, since we have no neutral gene calls in this df):
+    (
+      status != dominant_arm_status & status != dominant_cytoband_status
+    )
+    # Case 2) Gene call disagrees with a non-neutral cytoband call.
+    # Keep no matter what the arm status
+    | (
+      status != dominant_cytoband_status
+      & dominant_cytoband_status %in% c("gain", "loss")
+    )
+    # I think that captures everything we want. Cases we don't want include:
+    # gene & arm agree, but cytoband is neutral
+    # gene & cytoband agree
+    # all 3 agree
+  ) %>%
+  select(Kids_First_Biospecimen_ID = biospecimen_id,
+         gene_symbol,
+         status) %>%
+  # The `distinct()` function is needed to remove duplicates resulting from
+  # the band separation into multiple rows step above
+  distinct()
+
+# Display table
+filtered_consensus_gene_status
+```
+
+## Combine arm, cytoband, and gene-level status data
+
+```{r}
+# Rename each the dominant status columns of the data.frames
+# to be uniform for the binding rows step and filter out "neutral" calls
+consensus_seg_arm_status <- consensus_seg_arm_status %>%
+  filter(!(dominant_arm_status == "neutral")) %>%
+  rename(dominant_status = dominant_arm_status)
+
+filtered_consensus_cytoband_status <- filtered_consensus_cytoband_status %>%
+  filter(!(dominant_cytoband_status == "neutral")) %>%
+  rename(dominant_status = dominant_cytoband_status)
+
+# There are no "neutral" calls at the gene level so we do not need to filter
+# out those calls here
+filtered_consensus_gene_status <- filtered_consensus_gene_status %>%
+  rename(dominant_status = status)
+
+# For each of the datasets we're joining, we'll only keep these columns:
+cols_to_keep <- c("Kids_First_Biospecimen_ID", "dominant_status")
+```
+
+Combine into one long data frame
+
+```{r}
+# Combine the arm, cytoband, and gene status count data
+final_df <- bind_rows(
+  arm = select(consensus_seg_arm_status,
+               cols_to_keep,
+               region = chromosome_arm),
+  cytoband = select(filtered_consensus_cytoband_status,
+                    cols_to_keep,
+                    region = cytoband),
+  gene = select(filtered_consensus_gene_status,
+                cols_to_keep,
+                region = gene_symbol),
+  .id = "region_type"
+) %>%
+  # Reorder columns more sensibly
+  select(Kids_First_Biospecimen_ID,
+         status = dominant_status,
+         region,
+         region_type)
```

Comment:
This is not critical, but it might be worth sorting by ID, just to keep things in a somewhat reasonable order.
If we want to do even better, we could merge back in start and end positions for each region? Doing so might save a lot of redundant work in later scripts.

```suggestion
         region_type) %>%
    arrange(Kids_First_Biospecimen_ID)
```

Reply:
Gotcha, added this change in the last commit.